normandy/recipes/tests/__init__.py

```diff
@@ -156,6 +160,74 @@ class Meta:
     recipe = factory.SubFactory(RecipeFactory)
     identicon_seed = FuzzyIdenticonSeed()
     comment = FuzzyUnicode()
+    extra_filter_expression = factory.fuzzy.FuzzyChoice(["true", "false"])
+
+    @factory.lazy_attribute
+    def filter_object_json(self):
+        filters = [
+            ChannelFilterFactory(),
+            LocaleFilterFactory(),
+            CountryFilterFactory(),
+            StableSampleFilterFactory(),
+            BucketSampleFilterFactory(),
+        ]
+        return json.dumps(filters)
+
+
+class DictFactory(factory.Factory):
+    class Meta:
+        model = dict
+
+
+class ChannelFilterFactory(DictFactory):
+    class Params:
+        channel_objects = factory.LazyAttribute(lambda o: [ChannelFactory()])
+
+    type = "channel"
+    channels = factory.LazyAttribute(lambda o: [channel.slug for channel in o.channel_objects])
+
+
+class LocaleFilterFactory(DictFactory):
+    class Params:
+        locale_objects = factory.LazyAttribute(lambda o: [LocaleFactory()])
+
+    type = "locale"
+    locales = factory.LazyAttribute(lambda o: [locale.code for locale in o.locale_objects])
+
+
+class CountryFilterFactory(DictFactory):
+    class Params:
+        country_objects = factory.LazyAttribute(lambda o: [CountryFactory()])
+
+    type = "country"
+    countries = factory.LazyAttribute(lambda o: [country.code for country in o.country_objects])
+
+
+class FuzzySampleInputs(factory.fuzzy.FuzzyChoice):
+    def __init__(self, **kwargs):
+        super().__init__(
+            [
+                ["normandy.userId"],
+                ["normandy.userId", "42"],
+                ["normandy.userId", '"some study"'],
+                ["normandy.userId", "normandy.recipe.id"],
+                ["normandy.recipe.id", "normandy.userId"],
+            ]
+        )
+
+
+class StableSampleFilterFactory(DictFactory):
+    type = "stableSample"
+    input = FuzzySampleInputs()
+    rate = factory.fuzzy.FuzzyFloat(0.0, 1.0)
+
+
+class BucketSampleFilterFactory(DictFactory):
+    type = "bucketSample"
+    input = FuzzySampleInputs()
+    total = factory.fuzzy.FuzzyInteger(100, 10000)
+    start = factory.LazyAttribute(lambda o: factory.fuzzy.FuzzyInteger(0, o.total).fuzz())
+    count = factory.LazyAttribute(lambda o: factory.fuzzy.FuzzyInteger(0, o.total).fuzz())
```

Comment:
I doubt I'll change your mind but can you not see how complex this code looks? It looks really java'esque. Lambdas, factories, fuzzing, lazy evaluators. 
Sure, it works, but it doesn't read well. It might make sense right this minute when it's "hot" but simple code is more sustainable because it reads better when it's no longer hot. 

For example, if you need a recipe revision with a bucket sample filter object you could just do this:

```python
def test_bucket_sample_filter_object_expression():
    recipe = RecipeFactory()
    add_filter_object(recipe, {
        "type": "bucketSample", "start": 10, "count": 20, "total": 100, "input": ["normandy.userId"]
    })
    ...
```
It's more "verbose" but it's mentally parseable in a split second and because it's spelled out so simply you don't have to stop and think. What matters is the actual tests that code on the next line(s).

Yeah, there are advantages with frameworks and abstractions on top of all of that that can do integrity checks or randomness but is it needed? And at what price? God forbid, you have to do this amount of typing in a second unit test method, then just do it a second time. If you have to do it a third time, then you should probably stop and question why the test is potentially covering the same things twice.